### DISCUSSION TITLE
Cannot import 'NetworkXLabelGraphClusterer'

### DISCUSSION THREAD

**Discussion by ghiander** [scikit-multilearn/scikit-multilearn#277] [Q&A] [answered]
I am just trying to import NetworkXLabelGraphClusterer using scikit-multilearn==0.2.0 but I get this error as it is not in the package: "cannot import name 'NetworkXLabelGraphClusterer'"

I am just running the code you provide in "5.2.2. Estimating hyper-parameter k for embedded classifiers"

Best,

Giammy

**Comment by wizardwangbin** [comment]
I just meet the same problem. The solution is to import it as follow.

from skmultilearn.cluster.networkx import NetworkXLabelGraphClusterer

**Comment by oguerrer** [comment]
Hello,

I am having the same issue with `from skmultilearn.cluster import NetworkXLabelGraphClusterer`

It also comes up with 
`IGraphLabelGraphClusterer`, `GraphToolLabelGraphClusterer` and `StochasticBlockModel`

I tried the suggestion of using `from skmultilearn.cluster.networkx import NetworkXLabelGraphClusterer` but I get: `ModuleNotFoundError: No module named 'community'`

I have tried with versions 0.2.0 and 0.1.0 of scikit-multilearn. Any idea why this is happening?

Thanks

**Comment by wizardwangbin** [comment, marked as answer]
> Hello,
> 
> I am having the same issue with `from skmultilearn.cluster import NetworkXLabelGraphClusterer`
> 
> It also comes up with
> `IGraphLabelGraphClusterer`, `GraphToolLabelGraphClusterer` and `StochasticBlockModel`
> 
> I tried the suggestion of using `from skmultilearn.cluster.networkx import NetworkXLabelGraphClusterer` but I get: `ModuleNotFoundError: No module named 'community'`
> 
> I have tried with versions 0.2.0 and 0.1.0 of scikit-multilearn. Any idea why this is happening?
> 
> Thanks

You need to install the community package.  Download https://github.com/taynaud/python-louvain and install.

**Comment by oguerrer** [comment]
Thanks! This solves the issue.
